### ISSUE TITLE
CA1806 - False-negative for IQueryable Take()

### ISSUE THREAD

**Issue by pacoteinnov** [dotnet/roslyn-analyzers#7030]
### Analyzer

**Diagnostic ID**: [CA1806](https://learn.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/ca1806): `Do not ignore method results`

### Analyzer source

**NuGet Package**: [Microsoft.CodeAnalysis.NetAnalyzers](https://www.nuget.org/packages/Microsoft.CodeAnalysis.NetAnalyzers)

**Version**: 7.0.4 (Latest)

### Describe the bug

This code using IQueryable does not output CA1806 warning
```
IQueryable<int> queryable = (new int[] { 0, 1 }).AsQueryable();
queryable.Take(1);
```

But with IEnumerable it does
```
IEnumerable<int> enumerable= new int[] { 0, 1 };
enumerable.Take(1);
```

### Steps To Reproduce

Use the code above with latest analysis package.
`<AnalysisMode>All</AnalysisMode>` does not change the behavior.

### Expected behavior
Warning CA1806 with IQueryable methods that returns values.
### Actual behavior
No warning CA1806 with IQueryable methods that returns values.
